tfjs-backend-webgpu/src/kernels/reduce_webgpu.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {backend_util, util} from '@tensorflow/tfjs-core';
+import {ReduceInfo} from '../../../tfjs-core/src/ops/reduce_util';
+import {getCoordsDataType} from '../shader_preprocessor';
+import {computeDispatch} from '../webgpu_util';
+import {WebGPUProgram} from './webgpu_program';
+
+export class ReduceProgram implements WebGPUProgram {
+  outputShape: number[];
+  userCode: string;
+  dispatchLayout: {x: number[], y: number[]};
+  dispatch: [number, number, number];
+  workGroupSize: [number, number, number];
+  variableNames = ['x'];
+
+  constructor(reduceInfo: ReduceInfo, reduceType: 'max'|'min'|'sum') {
+    const inputShape = [reduceInfo.batchSize, reduceInfo.inSize];
+    const [outputShape, reduceShape] =
+        backend_util.computeOutAndReduceShapes(inputShape, [1]);
+    this.outputShape = outputShape.length === 0 ? [1] : outputShape;
+    const reduceSize = util.sizeFromShape(reduceShape);
+
+    const op = reduceType === 'min' ? '<' : '>';
+    const reductionFactor = 2;
+    const xMaxThreads = 1024;
```

Comment:
Just curious - have you tried different values of `xMaxThreads`? Does 1024 get the best performance?